### PR TITLE
Code Optimization and Refactoring

### PR DESCRIPTION
**Pull request by udohPixel**

### CODE REVIEW

**Review by kufre-okon — CHANGES_REQUESTED** [review]
Good work

File: answers/helpers/answerFilters.js [hunk 1]
```diff
@@ -18,6 +23,22 @@ const userFilters = {
       };
     }
 
+    // find by user
+    if (queryStr.username) {
+      const theUser = await User.findOne({
+        where: {
+          username: queryStr.username,
+        },
+      });
```
[the remark this answers is not in the document]
Comment (udohPixel) [reply answers/helpers/answerFilters.js:32]: Yes, it is. It was set as a unique field.

File: answers/helpers/answerFilters.js [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
+
+const sortArray = ANSWER_SORT_ARRAY;
 
 // app filters
-const userFilters = {
+const answerFilters = {
 
   // all answers filter function
-  filterItems: async (questionId, queryStr) => {
+  filterItems: async (QuestionId, queryStr) => {
```
[the remark this answers is not in the document]
Comment (udohPixel) [reply answers/helpers/answerFilters.js:14]: I changed it so I could easily recognise the request that was coming from `req.body` {I used QuestionId in this case} and `req.params` {I used questionId in this case}, without having to go to the controller each time to check.
Same goes for all others AnswerId, CommentId etc.

File: answers/helpers/answerFilters.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (udohPixel) [reply answers/helpers/answerFilters.js:32]: Okay boss. I will follow through in subsequent projects. Thanks. 👍

**Pull request merged**